service/parserprovider/gates.go

```diff
@@ -0,0 +1,106 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parserprovider
+
+import (
+	"context"
+	"flag"
+	"sort"
+	"strings"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+var _ flag.Value = (*gateFlagValue)(nil)
+
+// gateFlagValue implements the flag.Value interface and provides a mechanism for applying feature
+// gate statuses to a Registry.
+type gateFlagValue map[string]bool
+
+// String returns a string representing the gateFlagValue.
+func (f gateFlagValue) String() string {
+	var t []string
+	for k, v := range f {
+		if v {
+			t = append(t, k)
+		} else {
+			t = append(t, "-"+k)
+		}
+	}
+
+	// Sort the list of identifiers for consistent results
+	sort.Strings(t)
+	return strings.Join(t, ",")
+}
+
+// Set applies the gateFlagValue encoded in the input string.
+func (f gateFlagValue) Set(s string) error {
+	for id, state := range unmarshalGates(strings.Split(s, ",")) {
+		f[id] = state
+	}
+	return nil
+}
+
+type gatesFlagMapProvider struct{}
+
+// NewGatesMapProvider returns a MapProvider, that provides a config.Map from feature gate properties.
+//
+// The implementation reads --feature-gates flag(s) from the cmd and sets them in a config.Map
+func NewGatesMapProvider() MapProvider {
```

Comment:
Actually because the "service::gates" is a map, you can simply use "--set" flag now correct? No need to have this at all.